content/docs/grammar-language.md

```diff
@@ -0,0 +1,371 @@
+
+#### Keywords
+Keywords are meant to guide the parser in defining what type of object needs to be parsed.
+A keyword must match the regular expression `/\^?[_a-zA-Z][\w_]*/` and appear in the grammar file between single quotes `''`.
+
+Consider the following:
+```
+Person:
+    name=ID;
+
+City:
+    name=ID;
+```
+The parser will not be able to differentiate between the parsing of `Person` or `City` if it encounters a token matching the terminal rule `ID`. We can include *keywords* to remove the ambiguity:
+```
+Person:
+    'person' name=ID;
+
+City:
+    'city' name=ID;
+```
+Now the parser is able to differentiate between the two parser rules and will successfully parse objects of type `Person` and `City`.
+
+Parser rules can have many keywords:
+```
+Person:
+    'person' name=ID 'age' age=INT;
+```
+
+#### Assignments
+There are three different ways to assign an expression (right side) to a property (left side).
+
+1. `=` is used for assigning **only one element** to a property
+    ```
+    Person:
+        'person' name=ID
+    ```
+    Here, the property `name` will accept only one expression matching the terminal rule `ID`.
+
+2. `+=` is used to assign **multiple expressions** to a list property
+    ```
+    Paragraph:
+        'paragraph' (sentences+=STRING)+;
+    ```
+    Here, the property `sentences` will accept one or many expressions matching the terminal rule `STRING`. The expressions belong to a single group and must not be interrupted by other expressions. The parsing of a document containing:
+    ```
+    paragraph "The expression group " 3 "was interrupted"
+    ```
+    will throw an error since the `STRING` expressions are not continuous. It is however possible to interrupt and resume a sequence of expressions by using [hidden terminal symbols](#hidden-terminal-symbols):
+    ```
+    paragraph "expression one" /* comment */ "expression two"
+    ```
+    The above example will be successfully parsed.
+
+3. `?=` is used to assign a **value to a property of type boolean**. The value of the property of type `boolean` is set to `true` if the right part of the assignment is consumed by the parser.
+    ```
+    Employee:
+        'employee' name=ID (remote?='remote')?
+    ```
+    Here the value of the property `remote` will be set to true if the keyword `remote` appears in the document.
+
+#### Cross-reference
+With Langium, you can declare *cross-references* directly in the grammar. *Cross-reference* allows to reference an object of a given type. The syntax is:
+```
+property=[Type:TOKEN]
+```
+The `property` will be a reference to an object of type `Type` identified by the token `TOKEN`. If the `TOKEN` is omitted, the parser assumes that it is matching the terminal rule `ID`.
```

Comment:
Since we use a lot of functionality in this documentation which is only available in version 0.3, we should use the 0.3 behavior here as well:

```suggestion
The `property` will be a reference to an object of type `Type` identified by the token `TOKEN`. If the `TOKEN` is omitted, the parser will use the terminal or data type rule associated with the `name` assignment of the `Type` rule.
```